app/services/DataGenerator.scala

```diff
@@ -135,26 +138,67 @@ class DataGenerator @Inject()(auditEvents: AuditEvents,
+  def setValidatorCsv(sheetName:String)(implicit hc : HeaderCarrier, request: Request[_], messages: Messages): Either[Throwable, DataValidator] = {
+    Try {
+      ersValidationConfigs.getValidator(ersSheets(sheetName).configFileName)
+    } match {
+      case Success(validator) => Right(validator)
+      case Failure(e) =>
+        auditEvents.auditRunTimeError(e, "Could not set the validator", sheetName)(hc, request, ec)
+        Logger.error("[DataGenerator][setValidatorCsv] setValidator has thrown an exception, SheetName: " + sheetName + " Exception message: " + e.getMessage)
+        Left(ERSFileProcessingException(
+          "ers.exceptions.dataParser.configFailure",
+          Messages("ers.exceptions.dataParser.validatorError"),
+          optionalParams = Seq(sheetName)
+        ))
+    }
+  }
+
+  def identifyAndDefineSheet(filename: String, scheme: String)(implicit hc: HeaderCarrier, request: Request[_], messages: Messages): String = {
     Logger.debug("5.1  case 0 identifyAndDefineSheet  " )
-    val res = getSheet(data, scheme)
+    val sheetInfo = getSheet(filename, scheme)
     val schemeName = ersUtil.getSchemeName(scheme)._2
-    if (res.schemeType.toLowerCase == schemeName.toLowerCase) {
-      Logger.debug("****5.1.1  data contains data:  *****" + data)
-      data
+    if (sheetInfo.schemeType.toLowerCase == schemeName.toLowerCase) {
+      Logger.debug("****5.1.1  data contains data:  *****" + filename)
+      filename
     } else {
-      auditEvents.fileProcessingErrorAudit(res.schemeType, res.sheetName, s"${res.schemeType.toLowerCase} is not equal to ${schemeName.toLowerCase}")
-      Logger.warn(Messages("ers.exceptions.dataParser.incorrectSchemeType", res.schemeType.toUpperCase, schemeName.toUpperCase))
-      throw ERSFileProcessingException("ers.exceptions.dataParser.incorrectSchemeType",
-        Messages("ers.exceptions.dataParser.incorrectSchemeType", res.schemeType.toLowerCase, schemeName.toLowerCase),
-        optionalParams = Seq(ersUtil.withArticle(res.schemeType.toUpperCase), ersUtil.withArticle(schemeName.toUpperCase), res.sheetName))
+      auditEvents.fileProcessingErrorAudit(sheetInfo.schemeType, sheetInfo.sheetName, s"${sheetInfo.schemeType.toLowerCase} is not equal to ${schemeName.toLowerCase}")
+      Logger.warn(Messages("ers.exceptions.dataParser.incorrectSchemeType", sheetInfo.schemeType.toUpperCase, schemeName.toUpperCase))
+      throw ERSFileProcessingException(
+        "ers.exceptions.dataParser.incorrectSchemeType",
+        Messages("ers.exceptions.dataParser.incorrectSchemeType", sheetInfo.schemeType.toLowerCase, schemeName.toLowerCase),
+        optionalParams = Seq(ersUtil.withArticle(sheetInfo.schemeType.toUpperCase), ersUtil.withArticle(schemeName.toUpperCase), sheetInfo.sheetName))
     }
   }
 
-  def getSheet(sheetName:String, scheme:String)(implicit messages: Messages): SheetInfo = {
+   def identifyAndDefineSheetEither(informationOnInput: (SheetInfo, String))(
```

Comment:
Would it be better to name the two `identifyAndDefineSheet` functions ODS and CSV to make it clear which uses what?